sihl-queue/src/sihl_queue.mli

```diff
@@ -0,0 +1,87 @@
+(** [job] describes a job that can be dispatched. *)
+type job = Sihl.Contract.Queue.job =
+  { name : string
+  ; handle : string -> (unit, string) result Lwt.t
+  ; failed : string -> string -> string -> string -> unit Lwt.t
+  ; max_tries : int
+  ; retry_delay : Ptime.span
+  }
+
+(** [create_job handle ?max_tries ?retry_delay ?failed name] returns a job that
+    can be placed on the queue (dispatched) for later processing.
+
+    [handle] is the function that is called with the input when processing the
+    job. If an exception is raised, the exception is turned into [Error].
+
+    [max_tries] is the maximum times a job can fail. If a job fails [max_tries]
+    number of times, the status of the job becomes [Failed]. By default, a job
+    can fail [5] times.
+
+    [retry_delay] is the time span between two retries. By default, this value
+    is one minute.
+
+    [failed] is the error handler that is called when [handle] returns an error
+    or raises an exception. By default, this function does nothing. Use [failed]
+    to clean up resources or raise some error in a monitoring system in case a
+    job fails.
+
+    [name] is the name of the job, it has to be unique among all registered
+    jobs. *)
+val create_job
+  :  (string -> (unit, string) result Lwt.t)
+  -> ?max_tries:int
+  -> ?retry_delay:Ptime.span
+  -> ?failed:(string -> string -> string -> string -> unit Lwt.t)
```

Comment:
Pass error message and job instance directly.

Reply:
@jerben You missed this one. (GitHub hid it and you have to expand manually, pretty annoying)

Reply:
Fixed.